File: src/tf_pratice/src/tf2_pub_demo.cpp
#include "ros/ros.h"
#include "tf2_ros/static_transform_broadcaster.h"
#include "geometry_msgs/TransformStamped.h"
#include "tf2/LinearMath/Quaternion.h"

int main(int argc, char *argv[])
{
    setlocale(LC_ALL, "");
    ros::init(argc, argv, "tf2_pub_demo_node");
    ros::NodeHandle nh;

    tf2_ros::StaticTransformBroadcaster tf_broadcaster;
    geometry_msgs::TransformStamped transformStamped;
    transformStamped.header.stamp = ros::Time::now();
    
    transformStamped.header.frame_id = "base_link";
    transformStamped.child_frame_id = "lidar";

    transformStamped.transform.translation.x = 0.2;
    transformStamped.transform.translation.y = 0.0;
    transformStamped.transform.translation.z = 0.5;
    
    tf2::Quaternion q;
    q.setRPY(0, 0, 0);
    transformStamped.transform.rotation.x = q.getX();
    transformStamped.transform.rotation.y = q.getY();
    transformStamped.transform.rotation.z = q.getZ();
    transformStamped.transform.rotation.w = q.getW();

    tf_broadcaster.sendTransform(transformStamped);

    ros::spin();

    return 0;
}

File: src/tf_pratice/scripts/tf2_pub_demo.py
import rospy
import tf2_ros
import tf
from geometry_msgs.msg import TransformStamped

if __name__ == '__main__':
    rospy.init_node('tf2_pub_demo_py')
    broadcaster = tf2_ros.StaticTransformBroadcaster()

    transform = TransformStamped()
    transform.header.stamp = rospy.Time.now()
    transform.header.frame_id = "base_link"
    transform.child_frame_id = "lidar"

    transform.transform.translation.x = 0.2
    transform.transform.translation.y = 0.0
    transform.transform.translation.z = 0.5

    qtn = tf.transformations.quaternion_from_euler(0, 0, 0)
    
    transform.transform.rotation.x = qtn[0]
    transform.transform.rotation.y = qtn[1]
    transform.transform.rotation.z = qtn[2]
    transform.transform.rotation.w = qtn[3]


    broadcaster.sendTransform(transform)
    rospy.spin()